### ISSUE TITLE
Add 2 new commands to run.py

### ISSUE THREAD

**Issue by golnazads** [adsabs/ADSReferencePipeline#13]
We need to have the following functionality:
	python get_references -b <source_bibcode>  (generates prioritized, merged list)
	python get_references -b <source_bibcode> -a (returns all references with corresponding origin)

**Comment by ehenneken** [comment]
@golnazads I don't think we need this functionality in the pipeline, definitely not the first one. I have written a utility to do this type of extracting from the reference pipeline database, to support the integration into the Classic workflow. But in the (near?) future, all of this will go away when Honeycomb will take over all aggregating and merging of data.

**Comment by golnazads** [comment]
@edwin I have been working on this for a few days now and should have it done soon. I understand Honeycomb might eventually handle everything, but I do not think it can fully do what is needed here without the changes I am making to the database and pipeline. I am confident these updates will help bridge the gap until Honeycomb is fully in place.
@aaccomazzi

**Comment by golnazads** [comment]
For your information, the commands are now implemented, and I am confident that the pipeline is the right place to query the database and return the results. If there are any database schema changes that require updates to the query, the pipeline’s unit tests will fail, ensuring that the necessary updates are immediately addressed. This ensures the query is promptly fixed before the pipeline can continue functioning. However, if the query logic is placed outside the pipeline, such as in Honeycomb, there is no assurance that changes in the database will trigger updates to the external query. This disconnect could lead to broken queries and inconsistencies that are harder to detect and resolve.
@ehenneken @aaccomazzi

**Comment by golnazads** [comment]
I was not familiar with Honeycomb, so I did a search and found Honeycomb.io - I am wondering if this is the tool Edwin and Alberto have been referring to. After reading, I found that Honeycomb.io is actually an observability tool – it is meant to help devops monitor and understand how the systems are performing. It specializes in troubleshooting production issues, analyzing system performance, and understanding user behavior patterns. It is particularly good at helping users debug complex issues through its data analysis and tracing capabilities. However, since Edwin mentioned Honeycomb will "take over all aggregating and merging of data," I am wondering if he is actually referring to a different tool also called Honeycomb. In any case, from what I understand right now, I am staying by my original recommendation that queries should be in the pipeline. Having queries in the pipeline makes more sense because you can properly track changes through GitHub, test them, and manage them better. The pipeline also allows for better error handling, and logging. Most importantly, as I mentioned earlier, if there are any database schema changes, the pipeline's unit tests will fail immediately, ensuring necessary updates are addressed. If queries were outside the pipeline, there is no automatic way to detect when database changes break the queries. Devops can then use Honeycomb (if it is the observability tool) for what it is best at - monitoring how these queries are performing to help them spot any issues. This keeps ADS system organized and easier to maintain, with each tool doing what it is designed for.

@ehenneken @aaccomazzi @tjacovich @kelockhart

**Comment by golnazads** [comment]
OK, my curiosity took better of me, lol, searched team drive and came across multiple Honeycomb RFCs, read them. Now I understand that Honeycomb is to take over backoffice infrastructure, replacing the current celery/rabbitmq system. This actually reinforces my position about keeping the queries in the pipeline. While Honeycomb's Aggregator component is designed for matching and combining records from different sources, the pipeline still provides the best environment for managing database queries. My recommendation is to have each microservice/pipeline that integrates with the database supply Honeycomb with the formatted data that it needs. Honeycomb does not need to know the structure of all the tables in all the databases, it just needs to know which piece of data it needs, request it and combine it.
@tjacovich I would very much like to chat with you further about the Honeycomb architecture. I have a few questions and comments about the design. Let me know when you have some time to chat.

**Comment by golnazads** [comment]
@tjacovich I am actually very surprised that you have not reached out to me since I maintain both the link resolver and oracle databases. And going forward I will be maintaining both reference pipeline and planetary names pipeline databases. Let's chat.
